metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag } from '@opentelemetry/api';
+import {
+  DnsInstrumentation,
+  DnsInstrumentationConfig,
+} from '@opentelemetry/instrumentation-dns';
+import {
+  ExpressInstrumentation,
+  ExpressInstrumentationConfig,
+} from '@opentelemetry/instrumentation-express';
+import { GrpcInstrumentationConfig } from '@opentelemetry/instrumentation-grpc/build/src/types';
+import {
+  HttpInstrumentation,
+  HttpInstrumentationConfig,
+} from '@opentelemetry/instrumentation-http';
+import {
+  GraphQLInstrumentation,
+  GraphQLInstrumentationConfig,
+} from '@opentelemetry/instrumentation-graphql';
+import { GrpcInstrumentation } from '@opentelemetry/instrumentation-grpc';
+import { KoaInstrumentation } from '@opentelemetry/instrumentation-koa';
+// import {
+//   MySQLInstrumentation,
+//   MySQLInstrumentationConfig,
+// } from '@opentelemetry/instrumentation-mysql';
+import {
+  IORedisInstrumentation,
+  IORedisInstrumentationConfig,
+} from '@opentelemetry/instrumentation-ioredis';
+// import {
+//   MongoDBInstrumentation,
+//   MongoDBInstrumentationConfig,
+// } from '@opentelemetry/instrumentation-mongodb';
+import {
+  PgInstrumentation,
+  PgInstrumentationConfig,
+} from '@opentelemetry/instrumentation-pg';
+// import {
+//   RedisInstrumentation,
+//   RedisInstrumentationConfig,
+// } from '@opentelemetry/instrumentation-redis';
+import {
+  Instrumentation,
+  InstrumentationConfig,
+} from '@opentelemetry/instrumentation';
+
+type InstrumentationMapKeys =
```

Comment:
~~This can be done using `keyof InstrumentationMap` so you don't have to update in so many places~~

edit: I actually think all of these types can be compressed into a single place to manage